File: corporate-reporting/src/types/UserRole.ts
import { IEntity, Entity } from "./Entity";
import { IUser } from "./User";

export interface IUserRole extends IEntity {
    UserID: number;
    RoleID: number;
    User?: IUser;
    Role?: IEntity;
}

export class UserRole extends Entity implements IUserRole {
    public UserID = null;
    public RoleID = null;
}

File: ORB.Core/Models/UserRole.cs
using System;
using System.Collections.Generic;

namespace ORB.Core.Models
{
    public class UserRole : EntityWithEditor
    {
        public int UserID { get; set; }
        public int RoleID { get; set; }

        public virtual Role Role { get; set; }
        public virtual User User { get; set; }
    }
}
